Library/Formula/idris.rb

```diff
@@ -19,8 +18,18 @@ class Idris < Formula
   depends_on "cabal-install" => :build
   depends_on "gmp"
 
+  depends_on "libffi" => :recommended
+  depends_on "pkg-config" => :build if build.with? "libffi"
+
   def install
-    install_cabal_package
```

Comment:
It'd be useful to have varargs for appending to the final cabal invocation so that flags can be set for that final step. It's less common for packages to need special flags on the `--only-dependencies` step, so if someone needs that it's fine to expand the definition into a custom block like I did for this one.